File: Fire Safety/Assets/Bullshit Office Tile Kit/Scripts/SimpleDoorTrigger.cs
using UnityEngine;
using System.Collections;

public class SimpleDoorTrigger : MonoBehaviour {
	public Transform Door;
	public float OpenAngleAmount = 88.0f;
	public float SmoothRotation;	
	private string openDoorText = "Press F To Open The Door : Press E To Check The Door Handle";
	public GUIStyle InteractTextStyle;
		
	private bool init = false;
	private bool hasEntered = false;
	private bool doorOpen = false;
	private Vector3 startAngle;
	private Vector3 openAngle;	
	private Rect openDoorTextRect;
		
	void Start () {
		//Check if Door Game Object is properly assigned
		if(Door == null){
			Debug.LogError (this + " :: Door Object Not Defined!");
		}
		
		//Init Start and Open door angles
		startAngle = Door.eulerAngles;
		openAngle = new Vector3(startAngle.x, startAngle.y + OpenAngleAmount, startAngle.z);

		//Init Interact text Rect
		Vector2 textSize = InteractTextStyle.CalcSize(new GUIContent(openDoorText));
		openDoorTextRect = new Rect(Screen.width / 2 - textSize.x / 2, Screen.height - (textSize.y + 5), textSize.x, textSize.y);
		
		init = true;
	}
		
	void Update () {
		if(!init)
			return;
		
		HandleDoorRotation();
		HandleUserInput();	
	}
	
	void OnTriggerEnter(Collider other){
		if(other.tag == "Player"){
			hasEntered = true;
		}
	}
	
	void OnTriggerExit(Collider other){
		hasEntered = false;
	}
	
	void OnGUI(){
		if(!init || !hasEntered)
			return;
		
		GUI.Label(openDoorTextRect, openDoorText, InteractTextStyle);
	}
	
	void HandleDoorRotation(){
		if(!doorOpen)
			Door.rotation = Quaternion.Euler(Vector3.Slerp(Door.eulerAngles, startAngle, Time.deltaTime * SmoothRotation));
		else
			Door.rotation = Quaternion.Euler(Vector3.Slerp(Door.eulerAngles, openAngle, Time.deltaTime * SmoothRotation));
	}
	
	void HandleUserInput(){
		if (Input.GetKeyDown (KeyCode.F) && hasEntered) {
						doorOpen = !doorOpen;
		} else if (!doorOpen&&Input.GetKeyDown (KeyCode.E) && hasEntered) {
			Debug.Log ("Door Checked");
				}
		if (doorOpen) {
			openDoorText = "Press F To Close The Door";
				} else {
			openDoorText = "Press F To Open The Door : Press E To Check The Door Handle";
				}
	}
}

File: Fire Safety/Assets/Bullshit Office Tile Kit/Scripts/SimpleDoorTrigger.js
﻿public var Door : Transform;
public var OpenAngleAmount : float = 88.0f;
public var SmoothRotation : float;
	
private var init : boolean = false;
private var hasEntered : boolean = false;
private var doorOpen : boolean = false;
private var checkedDoor : boolean = false;
private var startAngle : Vector3;
private var openAngle : Vector3;
	
function Start() {
	// Check if Door Game Object is properly assigned
	if(Door == null){
		Debug.LogError (this + " :: Door Object Not Defined!");
	}
	
	// Init Start and Open door angles
	startAngle = Door.eulerAngles;
	openAngle = new Vector3(startAngle.x, startAngle.y + OpenAngleAmount, startAngle.z);
	
	init = true;
}
	
function Update() {
	if(!init)
		return;
	
	HandleDoorRotation();
	
	if(hasEntered) {
		HandleUserInput();	
	}
}

function OnTriggerEnter(col : Collider){
	// Only react to the player entering the trigger zone
	if(col.gameObject.tag == "Player"){
		hasEntered = true;
	}
}

function OnTriggerExit(col : Collider){
	hasEntered = false;
	StateManager.UpdateContextualState(ContextualState.None);
}

function HandleDoorRotation(){
	if(!doorOpen) {
		Door.rotation = Quaternion.Euler(Vector3.Slerp(Door.eulerAngles, startAngle, Time.deltaTime * SmoothRotation));
	}
	else {
		Door.rotation = Quaternion.Euler(Vector3.Slerp(Door.eulerAngles, openAngle, Time.deltaTime * SmoothRotation));
	}
}

function HandleUserInput(){
	if (Input.GetButtonDown("Interact")) {
		doorOpen = !doorOpen;
	}
	else if (Input.GetButtonDown("AltInteract")) {
		checkedDoor = true;
	}
	
	if(checkedDoor) {
		if(doorOpen) {
			StateManager.UpdateContextualState(ContextualState.CheckedDoorKnobAndDoorOpen);
		}
		else {
			StateManager.UpdateContextualState(ContextualState.CheckedDoorKnobAndDoorClosed);
		}
	}
	else if (doorOpen) {
		StateManager.UpdateContextualState(ContextualState.CanCloseDoor);
	}
	else {
		StateManager.UpdateContextualState(ContextualState.CanOpenDoor);
	}
}